### PR TITLE
Custom png io

### PR DESCRIPTION
**Pull request by feliwir**
- [x] Implement PNG chunk parsing
- [x] Support for RGB,RGBA, grayscale, grayscale with alpha
- [x] Support for paletted pngs
- [x] Support for CRC32 computation and validation
- [ ] Support for ADAM7 interlacing
- [ ] Support for bitdepths other than 8
- [ ] Support for encoding

### CODE REVIEW

File: libraries/libgraphic/png/PngReader.h [hunk 1]
```diff
@@ -13,12 +18,73 @@ class PngReader
...
+    {
+        return _pixels;
+    }
+
+    inline const DateTime &modified()
+    {
+        return _modified;
+    }
+
 private:
```
Comment (sleepy-monax) [inline libraries/libgraphic/png/PngReader.h:41]: Look like I never told you that I prefer the order of declarations to be private then public to public then private ;)

Comment (feliwir) [reply libraries/libgraphic/png/PngReader.h:41]: wat? :D

**Comment by feliwir** [comment]
Fixes #284

**Comment by sleepy-monax** [comment]
Ok let's merge this bad boy! :)

**Pull request merged**